### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import uniqueRandom = require('unique-random');
+const random = uniqueRandom(1, 10);
+
+console.log(random(), random(), random());
+//=> 5 2 6
+```
+*/
+declare function uniqueRandom(
+	minimumValue: number,
+	maximumValue: number
```
Comment (sindresorhus) [inline index.d.ts]: I'm starting to question whether this is actually an improvement. The `Value` part doesn't really add anything, as it's implied that it's a value. And using `maximum` over `max` is fine I guess, but `max` is even more common than `maximum` and is considered a standalone word, not just an abbreviation, but `min` is an abbreviation, so for consistency, I think `maximum` and `minimum` would be the best names. What do you think?

File: index.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
+module.exports = (minimumValue, maximumValue) => {
 	let prev;
-	return function rand() {
-		const num = Math.floor((Math.random() * (max - min + 1)) + min);
-		prev = (num === prev && min !== max) ? rand() : num;
+	return function random() {
+		const num = Math.floor(
+			(Math.random() * (maximumValue - minimumValue + 1)) + minimumValue
+		);
+		prev = num === prev && minimumValue !== maximumValue ? random() : num;
```
Comment (sindresorhus) [inline index.js]: `num` => `number`

Comment (sindresorhus) [reply index.js]: `prev` => `previous`

Comment (sindresorhus) [reply index.js]: Actually, in this case `previousValue` would be clearer... Naming is hard...

File: index.d.ts (hunk 1 above)
Comment (BendingBender) [reply index.d.ts]: I find your reasoning very conclusive, I was also very unsure about the renaming. Let's go with `minimum` and `maximum`.

**Pull request merged**